c10/core/TensorImpl.h

```diff
@@ -228,20 +229,62 @@ struct C10_API VariableVersion {
   c10::intrusive_ptr<VersionCounter> version_counter_;
 
  public:
+  enum Disabled { DISABLED };
```

Comment:
I think we are missing a whole paragraph of doc at the begining of version counter to explain what a DISABLED Version counter is.
Also, inference Tensor would be just one use case for this, so maybe we can make all the comments here independent of inference Tensors and just mention that they are one user of the DISABLED flag.